### PR TITLE
Add Hilbert-Schmidt innerproduct function

### PR DESCRIPTION
**Pull request by AdithyaSireesh**
## Description
- Added a function to compute the hilbert-schmidt inner product between two matrices
- Added tests to check if the function works as expected

## Questions
-  Am I missing any major tests
- Am I correct in assuming that the return type of the function is complex?
- Is there a good way to preview the math equations and references in the function's docstring?

## Status
-  [ ] Ready to go

### CODE REVIEW

File: toqito/state_metrics/hilbert_schmidt_inner_product.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    .. [WikHSO] Wikipedia: Hilbert-Schmidt operator.
+        https://en.wikipedia.org/wiki/Hilbert%E2%80%93Schmidt_operator
+    .. [Rigetti21] Forest Benchmarking (Rigetti).
+        https://github.com/rigetti/forest-benchmarking
+
+    :param A: An input matrix A.
+    :param B: An input matrix B.
+    :return: The Hilbert-Schmidt inner product between :code:`A` and :code:`B`.
+    """
+    hs_ip = np.trace(np.matmul(np.transpose(np.conj(A)), B))
```
Comment (vprusso) [inline toqito/state_metrics/hilbert_schmidt_inner_product.py:41]: I believe you can simplify this a tad:

```
return np.trace(A.conj().T @ B)
```

Comment (AdithyaSireesh) [reply toqito/state_metrics/hilbert_schmidt_inner_product.py]: Yes, that makes sense.

**Comment by vprusso** [comment]
Excellent. Thank you again for your contribution--I would love to see more PRs! Cheers.

**Comment by vprusso** [comment]
That's great, thank you @AdithyaSireesh ! Also, if you could give the project a "star" I hope that this would help with the overall visibility of the `toqito` project. Thank you again for your contribution!

**Pull request merged**